design/sequence/scheduler-low.puml

```diff
@@ -0,0 +1,86 @@
+@startuml
+
+box "Central Server" #Lightblue
+    participant ":Database"
+    participant ":Scheduler"
+    participant "job:Job"
+    participant ":Dispatcher"
+end box
+
+
+":Database" -> ":Scheduler" ++: reschedule()
+":Scheduler" -> ":Database" ++: get_changed_job()
+":Database" --> ":Scheduler" --: job: Job
+activate "job:Job"
+
+
+break job.get_status()==JobStatus.KILLED||\n job.get_status()==JobStatus.FINISHED
+    ":Scheduler" -> "job:Job" ++:check_was_running()
+    "job:Job" --> ":Scheduler" --: was_running: bool
+
+    alt was_running
+        alt job.get_status()==JobStatus.KILLED
+            ":Scheduler" -> ":Dispatcher" ++: kill_job(job)
+            ":Dispatcher" --> ":Scheduler" --
+        end
+        ":Scheduler" -> ":Database" ++: get_next_job(job.wc_id)
+        ":Database" --> ":Scheduler" --: new_job: Job
+        ":Scheduler" -> ":Dispatcher" ++ : dispatch_job(new_job)
+        ":Dispatcher" --> ":Scheduler" --:Message
+        ":Scheduler" -> ":Database"++: set_job_status(new_job, JobStatus.RUNNING)
```

Comment:
The Dispatcher should not access the Database afaik. Only the scheduler does that.

The Dispatcher sends a Message (this can be replaced with a bool or an int...) to tell the Scheduler that the Jobs has started successfully, the Scheduler then changes the state of the Job to "RUNNING"